app/packs/src/apps/mydb/elements/details/literature/DetailsTabLiteratures.js

```diff
@@ -222,16 +222,16 @@ export default class DetailsTabLiteratures extends Component {
       if (json.data && json.data.length > 0) {
         const data = json.data[0];
         const citation = new Cite(data);
-        this.setState((prevState) => ({
-          ...prevState,
-          literature: {
-            ...prevState.literature,
-            doi,
-            title: data.title || '',
-            year: (data && data.issued && data.issued['date-parts'][0]) || '',
-            refs: { citation, bibtex: citation.format('bibtex'), bibliography: json.format('bibliography') }
-          }
-        }));
+
+        this.setState((prevState) => {
+          let updatedLiterature = prevState.literature;
+          updatedLiterature.doi = doi;
+          updatedLiterature.title = data.title || '';
+          updatedLiterature.year = (data && data.issued && data.issued['date-parts'][0]) || '';
+          updatedLiterature.refs = { citation, bibtex: citation.format('bibtex'), bibliography: json.format('bibliography') };
+
+          return ({ literature: updatedLiterature });
+        });
```

Comment:
I think this modifies the `prevState.literature` object. This may not cause any issues but I'd prefer creating a new object, as in `{…prevState.literature, doi: doi, etc… }`